File: kh2_src/tabs/journal.py
from dash import Dash, html, dcc, callback, Input, Output, State, ALL
import kh2_src.kh2_utils as utils

from kh2_src.tabs.journal_tabs import *


@callback(
    Output("JournalDiv", "children"),
    Input("JournalTabs", "value"),
)
def __create_journal(tab):
    if tab == "Ansem Reports":
        return create_reports()
    if tab == "Character Files":
        return create_character_files()
    if tab == "Bestiary":
        return create_bestiary()

def create_journal():
    jtabs = dcc.Tabs(id="JournalTabs", value="Journal Flags")
    jtabs.children = [
        dcc.Tab(label="Journal Flags", value="Journal Flags"),
        dcc.Tab(label="Ansem Reports", value="Ansem's Reports"),
        dcc.Tab(label="Character Files", value="Character Files"),
        dcc.Tab(label="Bestiary", value="Bestiary"),
        dcc.Tab(label="Treasures", value="Treasures"),
        dcc.Tab(label="Puzzle Pieces", value="Puzzle Pieces"),
        dcc.Tab(label="Maps", value="Maps"),
        dcc.Tab(label="Missions", value="Missions"),
        dcc.Tab(label="Minigames", value="Minigames"),
        dcc.Tab(label="Synthesis Notes", value="Synthesis Notes"),
    ]
    return html.Div([
        jtabs,
        html.Div(id="JournalDiv"),
    ])

File: kh2_src/tabs/journal.js
import * as tabs from "./journal_tabs/index.js"

export function create_journal() {
    const kh2div = document.getElementById("kh2div");
    kh2div.innerHTML = `
    <div>
        <div>
            <h3>Entry:</h3>
            <select id="kh2jtabs">
                <option value="Journal Flags">Journal Flags</option>
                <option value="Ansem Reports">Ansem Reports</option>
                <option value="Character Files">Character Files</option>
                <option value="Bestiary">Bestiary</option>
                <option value="Treasures">Treasures</option>
                <option value="Puzzle Pieces">Puzzle Pieces</option>
                <option value="Maps">Maps</option>
                <option value="Missions">Missions</option>
                <option value="Minigames">Minigames</option>
                <option value="Synthesis Notes">Synthesis Notes</option>
            </select>
        </div>
        <div id="kh2jdiv"></div>
    </div>`;
    const kh2jtabs = document.getElementById("kh2jtabs");
    const kh2jdiv = document.getElementById("kh2jdiv");
    function tab_sel() {
        switch (kh2jtabs.value) {
            /*
            case "Journal Flags": {
                tabs.create_flags();
                break;
            }
            case "Ansem Reports": {
                tabs.create_reports();
                break;
            }
            case "Character Files": {
                tabs.create_character_files();
                break;
            }
            */
            case "Bestiary": {
                tabs.create_bestiary();
                break;
            }
            /*
            case "Treasures": {
                tabs.create_treasures();
                break;
            }
            */
            case "Minigames": {
                tabs.create_minigames();
                break;
            }
            /*
            case "Synthesis Notes": {
                tabs.create_synthesis_notes();
                break;
            }
            */
            default: {
                kh2jdiv.innerHTML = "";
                break;
            }
        }
    }
    kh2jtabs.addEventListener("change", tab_sel);
    tab_sel();
}
